test/softmax/test_softmax_interface.cpp

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2022, Advanced Micro Devices, Inc. All rights reserved.
+
+#include <stdexcept>
+#include <vector>
+
+#include "gtest/gtest.h"
+#include "test_softmax_util.hpp"
+
+class TestSoftmaxInterface : public ::testing::Test
+{
+    protected:
+    template <ck::index_t Rank, ck::index_t NumReduceDims>
+    using SoftmaxInstance =
+        ck::DeviceSoftmaxInstanceWrapper<Rank, NumReduceDims, 256, 1, 256, 1, 8, 1, 8, 8>;
+};
+
+TEST_F(TestSoftmaxInterface, IncorrectReduceDims)
+{
+    std::vector<ck::index_t> lengths{2, 128, 1536};
+    std::vector<ck::index_t> strides{128 * 1536, 1536, 1};
+
+    EXPECT_THROW((SoftmaxInstance<3, 1>{}.IsSupported(lengths, strides, {-1})), std::runtime_error);
```

Comment:
Where does DeviceSoftmaxImpl throw? Shouldn't we be expecting a boolean here? 

Assuming it does throw from somewhere, I suggest replacing EXPECT_THROW with EXPECT_FALSE for all tests checking against IsSupportedArgument() and avoid throwing exception in IsSupportedArgument(). The softmax profiler code (as well as most other ops' profiler codes in CK) does not expect to handle exceptions. 

There is a valid case when it is more desirable to silently skip unsupported kernels. Such as the case that we want to benchmark a specific GEMM size against a pool of kernel instances that may or may not support padding. Or likewise in softmax that an arbitrary input tensor that doesn't satisfy the condition of aligned vectorized load/store (SrcScalarPerVector) shouldn't cause the profiler to come to a full stop.

Reply:
@rosenrodt I understand your concerns regarding throwing an exception. However I couldn't find any better solution to reject creating an `Argument` for DeviceOp when the user pass ie. reduction axis which are out of input tensor bounds. Take a look here: https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/4b4b66756cecc5b57b4453f9a5015c4a603d54b3/include/ck/tensor_operation/gpu/device/device_reduce_multiblock.hpp#L223-L245

Previously without those checks the user could run such illformed case since in this place it isn't checked correctly: https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/develop/include/ck/tensor_operation/gpu/device/device_reduce_common.hpp#L75-L105 - this is the incorrect use of `assert` which is in Release build mode a no op.

Reply:
Thanks for explanation. I see what's going on here—It's not IsSupportedArgument() who throws, but MakeArgument().

Two scenarios where we can't run a problem size against specific kernel type:
1. **Unsupported**: problem may match at least one kernel, because some kernels have their own implementation details which prevent all kernels from running correctly (eg GEMM MNK padding, ScalarPerVector)
2. **Incorrect**: problem does not match any kernel at all (reducing 2 dimensions when kernel only supports 1 dimension reduction)

The cases that fall into incorrect scenario should throw an exception as it's obviously user error. The cases that fall into unsupported scenario is not user error. All look good to me now.